trie-db/src/lib.rs

```diff
@@ -377,52 +403,69 @@ where
 /// Trait with definition of trie layout.
 /// Contains all associated trait needed for
 /// a trie definition or implementation.
-pub trait TrieLayout {
+/// Structure implementing this trait can define contextual behavior
+/// for trie, but should be small (clone cost should be insignifiant).
+pub trait TrieLayout: Default + Clone {
```

Comment:
why do we bound `Default` here, isn't it error prone ?

I see for instance that `SecTrieDB::new` makes use of the default layout, I don't really know SecTrieDB but couldn't all those `new` which uses default layout be misused, and were actually meant to be called with a specific layout ?

Anyhow as far as I understand the code for now I feel like making the max_inline_value a const could solve most of this ? But maybe I miss the actual reason for all of this.

Reply:
Yes default in trie crate is old layout while default in substrate pr is new layout.

> Anyhow as far as I understand the code for now I feel like making the max_inline_value a const could solve most of this ? But maybe I miss the actual reason for all of this.

Reach the same conclusion (at first we got the layout behaviour from a state in the trie, so we could not have const), will change it today and come back to you if there is an unexpected issue.